### ISSUE TITLE
Dynamic Columns doesn't allow to insert an entity

### ISSUE THREAD

**Issue by iremezoff** [linq2db/linq2db#1268]
Hello!

I use the Dynamic Columns feature through fluent mapping builder as shown below:
```c#
public class MyClass
{
        public string Name { get; set; }
}

var ms = new MappingSchema();
var builder = ms.GetFluentMappingBuilder();
builder.Entity<MyClass>().Property(x=>Sql.Property<bool>(x, "IsDeleted"));

using (var db = new AffiliatesDatabase())
{
        db.AddMappingSchema(ms);
        var mc = new MyClass() {Name="test"};
        
        db.InsertWithIdentity(n);
}
```
The database representation has the required column with proper type (in my case, it's bit in sql server).
When my program hits insert operation the exception is thrown:
```
System.ArgumentException: ''IsDeleted' is not a member of type 'Test.MyClass''
```

When I use 'HasSkipOnInsert' the issue goes away but I lose possibility to set the value. Also **Update** operation catches the same issue.

Everything goes properly only on retrieving data, and even **IsDeleted** value is put into Dictionary field with **DynamicColumnsStore** attribute.

Is it possible to use dynamic columns with inserting?

### Environment details
linq2db version: *2.2.0*
Database Server: Sql Server 14
Database Provider: .net sql data provider
Operating system: win10
Framework version: .net core 2.0

**Comment by sdanyliv** [comment]
@iremezoff, ok. I have reproduced bug.
Fix is obvious but you have to remember that `"IsDeleted", false` should be in DynamicColumnsStore.  Otherwise insert will fail.

**Comment by sdanyliv** [comment]
It was originally developed for dynamic queries and databases that varies column.
But i agree that this needs some slight improvements here.
Will check how it fits with library design.

Ususally we do not care about model perfection - it represents real database how fileds are defined.
But how we work in that case:

**Comment by sdanyliv** [comment]
Fixed in d6fbcbaf30d3433e7fa982c27480e87561f2c40e
It is allow for now to get dynamic value even there is no Item in dictionary.
